lib/charms/acme_client_operator/v0/acme_client.py

```diff
@@ -0,0 +1,207 @@
+        if not self.unit.is_leader():
+            return
+
+        if not self._container.can_connect():
+            self.unit.status = WaitingStatus("Waiting for container to be ready")
+            event.defer()
+            return
+
+        try:
+            csr = x509.load_pem_x509_csr(event.certificate_signing_request.encode())
+            subject_value = csr.subject.get_attributes_for_oid(NameOID.COMMON_NAME)[0].value
+            if isinstance(subject_value, bytes):
+                subject = subject_value.decode()
+            else:
+                subject = subject_value
+        except Exception:
+            logger.exception("Bad CSR received, aborting")
+            return
+
+        self._container.push(
+            path=self._csr_path, make_dirs=True, source=event.certificate_signing_request.encode()
+        )
+
+        logger.info("Received Certificate Creation Request for domain %s", subject)
+        process = self._container.exec(
+            self.cmd, timeout=300, working_dir="/tmp", environment=self.plugin_config
+        )
+        try:
+            stdout, error = process.wait_output()
+            logger.info(f"Return message: {stdout}, {error}")
+        except ExecError as e:
+            self.unit.status = BlockedStatus("Error getting certificate. Check logs for details")
+            logger.error("Exited with code %d. Stderr:", e.exit_code)
+            for line in e.stderr.splitlines():  # type: ignore
+                logger.error("    %s", line)
+            return
+
+        chain_pem = self._container.pull(path=f"{self.certs_path}{subject}.crt")
+        certs = []
+        for cert in chain_pem.read().split("\n\n"):
+            certs.append(cert)
+        self.tls_certificates.set_relation_certificate(
+            certificate=certs[0],
+            certificate_signing_request=event.certificate_signing_request,
+            ca=certs[-1],
+            chain=list(reversed(certs)),
+            relation_id=event.relation_id,
+        )
+
+    @property
+    @abstractmethod
+    def cmd(self) -> list[str]:
+        """Command to run to get the certificate.
+        Implement this method in your charm to return the command that will be used to get the
+        certificate.
+        Example
+        ```
+        @property
+        def cmd(self):
+            return [
+                "lego",
+                "--email",
+                self._email,
+                "--accept-tos",
+                "--csr",
+                "/tmp/csr.pem",
+                "--server",
+                self._server,
+                "--dns",
+                "namecheap",
+                "run",
+            ]
+        ```
+
+        Returns:
+            list[str]: Command and args to run.
+        """
+
+    @property
+    @abstractmethod
+    def certs_path(self) -> str:
+        """Path to the certificates.
+        Implement this method in your charm to return the path to the obtained certificates.
```

Comment:
It's still unclear to me what this path is used for. Do I have to place something there? Is something going to be placed for me here?